intro-HPC/ch_MATLAB.tex

```diff
@@ -0,0 +1,117 @@
+\fi
+
+\section{How to compile MATLAB code}
+
+Compiling MATLAB code can only be done from the login nodes, because only login
+nodes can access the server, workernodes on clusters cannot.
+
+To access the MATLAB compiler, the MATLAB module should be loaded first. Make sure
+you are using the same MATLAB version to compile and to run the compiled MATLAB
+program.
+
+\begin{prompt}
+%\shellcmd{module avail MATLAB}%
+----------------------/apps/gent/CO7/sandybridge/modules/all ----------------------
+   MATLAB/2016b    MATLAB/2017b    MATLAB/2018a (D)
+%\shellcmd{module load MATLAB/2018a}
+\end{prompt}
+
+After loading the module, the \verb|mcc| command can be used. To get help on
+\verb|mcc|, you can run \verb|mcc -?|.
+
+To compile a single file, the \verb|-m|
+flag is used (the \verb|-v| flag means verbose ouput).
+To show how \verb|mcc| can be used, we use the \verb|magicsquare| example
+that comes with MATLAB.
+
+\begin{prompt}
+%\shellcmd{cp \$EBROOTMATLAB/extern/examples/compiler/magicsquare.m .}%
+%\shellcmd{mcc -mv magicsquare.m}%
+Opening log file:  %\homedir/java.log.34090%
+Compiler version: 6.6 (R2018a)
+Dependency analysis by REQUIREMENTS.
+Parsing file "%\homedir/magicsquare.m%"
+	(Referenced from: "Compiler Command Line").
+Deleting 0 temporary MEX authorization files.
+Generating file "%\homedir/readme.txt%".
+Generating file "run\_magicsquare.sh".
+\end{prompt}
+
+\subsection{Libraries}
+
+To \emph{compile} a MATLAB library, the \verb|-l| flag should be used.
+
+To compile a MATLAB program that \emph{needs a library}, you can use the
+\texttt{-I \emph{library\_path}} flag. This will tell the compiler to also
+look for files in \texttt{\emph{library\_path}}.
+
+It's also possible to use the \texttt{-a \emph{path}} flag. That will result in
+all files under the \texttt{\emph{path}} getting added to the final executable.
+
+\subsection{Memory issues during compilation}
+
+If you are seeing Java memory issues during the compilation of your MATLAB program
+on the login nodes, consider tweaking the default maximum heap size (128M) of Java
+using the \verb|_JAVA_OPTIONS| environment variable with:
+
+\begin{prompt}
+%\shellcmd{export \_JAVA\_OPTIONS="-Xmx64M"}%
+\end{prompt}
+
+The MATLAB compiler spawns multiple Java processes, and because of the default memory
+limits that are in effect on the login nodes, this might lead to a crash of the compiler
+if it's trying to create to many Java processes. If we lower the heap size, more
+Java processes will be able to fit in memory.
+
+\section{Multithreading}
+
+MATLAB can only use the cores in a single
+workernode (unless the Distributed Computing toolbox is used, see
+\url{https://nl.mathworks.com/products/distriben.html}).
+
+The amount of workers used by MATLAB for the parallel toolbox can be controlled
+via the \verb|parpool| function: \verb|parpool('cluster1', 16)| will run the
+\verb|cluster1| profile with 16 workers.
```

Comment:
This should be fleshed out more: missing ref, missing warning about default # workers, ...